python/src/ops.cpp

```diff
@@ -3587,6 +3587,36 @@ void init_ops(nb::module_& m) {
             array: The output containing elements selected from
             ``x`` and ``y``.
       )pbdoc");
+  m.def(
+      "nan_to_num",
+      [](const ScalarOrArray& a,
+         const ScalarOrArray& nan,
+         const ScalarOrArray& posinf,
+         const ScalarOrArray& neginf,
```

Comment:
I think the type should be something like a variant of monostate (None), Python float, or Python int.

Reply:
Concretely I think the type `std::variant<std::monostate, int, float>` is what we want here. Then if it's the monostate (None) you can set the defaults accordingly inside the function.